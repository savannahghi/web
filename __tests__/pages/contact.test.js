import { render, fireEvent, cleanup, waitFor } from '@testing-library/react';
import { shallow } from 'enzyme';
import preloadAll from 'jest-next-dynamic';
import { act } from 'react-dom/test-utils';
import ContactForm from '../../pages/contact';

let wrapper;
let nameInput;
let emailInput;
let messageInput;
let submitButton;

// Create initial props that get passed into the component
const initialProps = {
    location: {
        state: {
            from: {
                pathname: '/contact'
            }
        }
    }
};

// Unit testing
describe('Unit tests', () => {
    // await preloadAll();

    // what to do before each test
    beforeEach(() => {
        // Render the login form component, pass in props. (Shallow method renders the component without its children, good for unit tests.)
        wrapper = shallow(<ContactForm {...initialProps} />);
        nameInput = wrapper.find('#nameInput');
        emailInput = wrapper.find('#emailInput');
        messageInput = wrapper.find('#messageInput');
        submitButton = wrapper.find('#submitButton');
    });

    // what to do after each test
    afterEach(() => {
        jest.clearAllMocks();
    });

    // UI Integrity test
    it('should match the snapshot', () => {
        // snapshots are text references of the html of the rendered component.
        expect(wrapper.html()).toMatchSnapshot();
    });

    // name field
    it('should have a name inputs', () => {
        expect(nameInput.length).toEqual(1);
    });

    it('should have the expected props on the name field', () => {
        expect(nameInput.props()).toStrictEqual({
            id: 'nameInput',
            className: 'form-control',
            name: 'name',
            type: 'text'
        });
    });

    // email field
    it('should have an email field', () => {
        expect(emailInput.length).toEqual(1);
    });

    it('should have the expected props on the email field', () => {
        expect(emailInput.props()).toStrictEqual({
            id: 'emailInput',
            className: 'form-control',
            name: 'email',
            type: 'email'
        });
    });

    // message field
    it('should have a message field', () => {
        expect(messageInput.length).toEqual(1);
    });

    it('should have the expected props on the email field', () => {
        expect(messageInput.props()).toStrictEqual({
            id: 'messageInput',
            className: 'form-control',
            name: 'message',
            rows: '4',
            type: 'text'
        });
    });

    // submit button
    it('should have a submit button', () => {
        expect(submitButton.length).toEqual(1);
    });

    it('should have the expected props on the button', () => {
        expect(submitButton.props()).toEqual({
            id: 'submitButton',
            type: 'submit',
            'aria-label': 'Submit Button',
            className: 'btn btn-secondary font-weight-bold text-capitalize mb-4',
            children: 'Submit'
        });
    });
});

// Integrations Testing
describe('Integrations tests', () => {
    beforeEach(() => {
        // Render the login form component, pass in props. (render method renders the component with its children, good for integrations tests, uses react-test-library.)
        const { getByLabelText, getByText } = render(<ContactForm {...initialProps} />);
        nameInput = getByLabelText(/Name/i);
        emailInput = getByLabelText(/Email/i);
        messageInput = getByLabelText(/Message/i);
        submitButton = getByText('Submit');
    });

    afterEach(cleanup);

    it('name text change in onChange event', () => {
        expect(nameInput.value).toBe('');

        fireEvent.change(nameInput, { target: { value: 'John' } });

        expect(nameInput.value).toBe('John');
    });

    it('email text change in onChange event', () => {
        expect(emailInput.value).toBe('');

        fireEvent.change(emailInput, { target: { value: 'support@savannahghi.org' } });

        expect(emailInput.value).toBe('support@savannahghi.org');
    });

    it('message text change in onChange event', () => {
        expect(messageInput.value).toBe('');

        fireEvent.change(messageInput, {
            target: { value: 'This is a test message from Savannah Global Health Web-app' }
        });

        expect(messageInput.value).toBe(
            'This is a test message from Savannah Global Health Web-app'
        );
    });
});

test('Work Component', async () => {
    await preloadAll();

    const { getByText } = render(<ContactForm />);

    const lazyContent = await waitFor(() =>
        getByText('Or Reach Out to Us Via Our Social media handles')
    );
    expect(lazyContent).toBeInTheDocument();
});

describe('test dynamic imports', () => {
    act(() => {
        const component = shallow(<ContactForm />);
        expect(component.find('button')).toHaveLength(5);
        component.find('button').at(0).simulate('click').first;
    });
});
