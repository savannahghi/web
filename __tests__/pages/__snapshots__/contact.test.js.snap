// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit tests should match the snapshot 1`] = `"<div><main class=\\"sghi_container\\"><nav id=\\"navbar\\" class=\\"navbar sticky-top navbar-expand-lg navbar-light bg-light w-100\\"><div class=\\"container\\"><a class=\\"navbar-brand\\" href=\\"/\\"><div style=\\"display:inline-block;max-width:100%;overflow:hidden;position:relative;box-sizing:border-box;margin:0\\"><div style=\\"box-sizing:border-box;display:block;max-width:100%\\"><img style=\\"max-width:100%;display:block;margin:0;border:none;padding:0\\" alt=\\"\\" aria-hidden=\\"true\\" role=\\"presentation\\" src=\\"data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTUwIiBoZWlnaHQ9IjgwIiB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHZlcnNpb249IjEuMSIvPg==\\"/></div><img alt=\\"Brand Logo\\" src=\\"data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7\\" decoding=\\"async\\" style=\\"visibility:hidden;position:absolute;top:0;left:0;bottom:0;right:0;box-sizing:border-box;padding:0;border:none;margin:auto;display:block;width:0;height:0;min-width:100%;max-width:100%;min-height:100%;max-height:100%\\"/></div></a><button class=\\"navbar-toggler\\" type=\\"button\\" data-mdb-toggle=\\"collapse\\" data-mdb-target=\\"#navbarSupportedContent\\" aria-controls=\\"navbarSupportedContent\\" aria-expanded=\\"false\\" aria-label=\\"Toggle navigation\\"><i class=\\"fas fa-bars\\"></i></button><div class=\\"collapse navbar-collapse text-center menu-links\\" id=\\"navbarSupportedContent\\"><ul class=\\"navbar-nav ms-auto mx-auto mb-2 mb-lg-0\\"><li class=\\"nav-item\\"><a class=\\"nav-link\\" aria-current=\\"page\\" href=\\"/\\">Home</a></li><li class=\\"nav-item\\"><div class=\\"nav-link\\"><a href=\\"/#our-story\\">Our Story</a></div></li><li class=\\"nav-item\\"><div class=\\"nav-link\\"><a href=\\"/#our-work\\">Our Work</a></div></li><li class=\\"nav-item\\"><div class=\\"nav-link\\"><a href=\\"/#dna\\">Our DNA</a></div></li><li class=\\"nav-item\\"><div class=\\"nav-link\\"><a href=\\"/#partners\\">Partners</a></div></li><li class=\\"nav-item\\"><div class=\\"nav-link\\"><a href=\\"/products\\">Products</a></div></li><li class=\\"nav-item\\"><div class=\\"nav-link\\"><a href=\\"/team\\">Teams</a></div></li><li class=\\"nav-item\\"><div class=\\"nav-link active\\"><a href=\\"/contact\\">Contact</a></div></li></ul></div></div></nav><main class=\\"container\\"><div class=\\"row\\"><div class=\\"col-12\\"><p class=\\"page-title text-center pt-5\\"><span class=\\"page-title-second\\">Contact </span>Us </p></div><div class=\\"col-12\\"><div class=\\"\\"><div class=\\"mb-5\\"><form method=\\"post\\" class=\\"p-3\\"><div class=\\"mb-4\\"><label class=\\"form-label\\" for=\\"nameInput\\">Name </label><input type=\\"text\\" name=\\"name\\" id=\\"nameInput\\" class=\\"form-control\\"/><span class=\\"error_message\\"></span></div><div class=\\"mb-4\\"><label class=\\"form-label\\" for=\\"emailInput\\">Email </label><input type=\\"email\\" name=\\"email\\" id=\\"emailInput\\" class=\\"form-control\\"/><span class=\\"error_message\\"></span></div><div class=\\"mb-4\\"><label class=\\"form-label\\" for=\\"messageInput\\">Message</label><textarea name=\\"message\\" type=\\"text\\" class=\\"form-control\\" id=\\"messageInput\\" rows=\\"4\\"></textarea><span class=\\"error_message\\"></span></div><div class=\\"contact-submit-btn \\"><button aria-label=\\"Submit Button\\" type=\\"submit\\" id=\\"submitButton\\" class=\\"btn btn-secondary font-weight-bold text-capitalize mb-4\\">Submit</button></div><div class=\\"text-center\\"><p>Or Reach Out to Us Via Our Social media handles</p><button type=\\"button\\" aria-label=\\"LinkedIn Button\\" class=\\"btn btn-lg btn-secondary btn-floating mx-1\\"><a target=\\"_blank\\" aria-label=\\"LinkedIn\\" href=\\"https://www.linkedin.com/company/savannahglobalhealth\\" rel=\\"noopener noreferrer\\"><i class=\\"fab fa-linkedin-in\\"></i></a></button><button type=\\"button\\" aria-label=\\"Twitter Button\\" class=\\"btn btn-lg btn-secondary btn-floating mx-1\\"><a target=\\"_blank\\" aria-label=\\"Twitter\\" href=\\"https://twitter.com/SGHI76989573\\" rel=\\"noopener noreferrer\\"><i class=\\"fab fa-twitter\\"></i></a></button><button type=\\"button\\" aria-label=\\"Facebook Button\\" class=\\"btn btn-lg btn-secondary btn-floating mx-1\\"><a target=\\"_blank\\" aria-label=\\"Facebook\\" href=\\"https://fb.me/SavannahGlobalHealth\\" rel=\\"noopener noreferrer\\"><i class=\\"fab fa-facebook-f\\"></i></a></button></div></form></div></div></div><div class=\\"col-12 py-5\\">Loading Map...</div></div></main></main><div><footer class=\\"text-white text-center text-lg-start\\"><div class=\\"container p-4\\"><div class=\\"row\\"><div class=\\"col-lg-5 col-md-12 mb-4 mb-md-0 p-3\\"><div><div><div style=\\"display:inline-block;max-width:100%;overflow:hidden;position:relative;box-sizing:border-box;margin:0\\"><div style=\\"box-sizing:border-box;display:block;max-width:100%\\"><img style=\\"max-width:100%;display:block;margin:0;border:none;padding:0\\" alt=\\"\\" aria-hidden=\\"true\\" role=\\"presentation\\" src=\\"data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMjAwIiBoZWlnaHQ9IjEwMCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4=\\"/></div><img alt=\\"Brand Logo\\" src=\\"data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7\\" decoding=\\"async\\" style=\\"visibility:hidden;position:absolute;top:0;left:0;bottom:0;right:0;box-sizing:border-box;padding:0;border:none;margin:auto;display:block;width:0;height:0;min-width:100%;max-width:100%;min-height:100%;max-height:100%\\"/></div></div><p>About the SGHI</p><p>SGHI is a development ecosystem builder and optimizer that supports philanthropy and community effort to leverage investment and deliver lasting impact</p></div></div><div class=\\"col-lg-2 col-md-6 mb-4 mb-md-0 pt-5 p-3\\"><div><p>Additional links</p><ul><li><a href=\\"/\\">Services</a></li><li><a href=\\"/\\">Contact us</a></li><li><a href=\\"/\\">Why we exist</a></li><li><a href=\\"/\\">SGHI MISSION AND VISION</a></li></ul></div></div><div class=\\"col-lg-5 col-md-6 col-sm-12 mb-4 mb-md-0 pt-5 ml-5\\"><div><p>Latest tweets</p><ul><li><div class=\\"row\\"><div class=\\"col-2\\"><i class=\\"fab fa-twitter\\"></i></div><div class=\\"col-10\\"><a href=\\"/\\">RT @wpklik: Here is our selection of 20 #Worldrevolutionaries @edgethemes @selectthemes @mi…</a></div></div></li><li><div class=\\"row\\"><div class=\\"col-2\\"><i class=\\"fab fa-twitter\\"></i></div><div class=\\"col-10\\"><a href=\\"/\\">RT @wpklik: Here is our selection of 20 #Worldrevolutionaries @edgethemes @selectthemes @mi…</a></div></div></li></ul></div></div><div class=\\"col-lg-5 col-md-6 mb-4 mb-md-0\\"><div><p>Connect with us</p><div><div class=\\"row\\"><div class=\\"col-1 mx-2\\"><a target=\\"_blank\\" aria-label=\\"LinkedIn\\" href=\\"https://www.linkedin.com/company/savannahglobalhealth\\" rel=\\"noopener noreferrer\\"><i class=\\"fab fa-linkedin-in fa-2x\\"></i></a></div><div class=\\"col-1 mx-2\\"><a target=\\"_blank\\" aria-label=\\"Twitter\\" href=\\"https://twitter.com/SGHI76989573\\" rel=\\"noopener noreferrer\\"><i class=\\"fab fa-twitter fa-2x\\"></i></a></div><div class=\\"col-1 mx-2\\"><a target=\\"_blank\\" aria-label=\\"Facebook\\" href=\\"https://fb.me/SavannahGlobalHealth\\" rel=\\"noopener noreferrer\\"><i class=\\"fab fa-facebook-f fa-2x\\"></i></a></div></div></div></div></div></div></div></footer></div></div>"`;
